### ISSUE TITLE
chunk options with non-breaking space does not work and are not shown as invalid

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#2681]
### Discussed in https://github.com/quarto-dev/quarto-cli/discussions/2678

<div type='discussions-op-text'>

<sup>Originally posted by **cbrnr** September 30, 2022</sup>
Recently, it took me longer than I'd like to admit to find out why the following code chunk gets evaluated despite setting the corresponding option:

````
---
title: Test
---

```{r}
#| eval: false
cat("Hello")
```
````

If you want to reproduce, copy this example to a file `test.qmd` and preview it with `quarto preview test.qmd`.

In fact, the cell is evaluated because I accidentally used a non-breaking space (U+00A0, <kbd>⌥</kbd><kbd>Space</kbd> on a Mac) instead of a normal space after the `#|` characters. I now turned on `"editor.unicodeHighlight.includeComments": true` and `"editor.unicodeHighlight.nonBasicASCII": true` in Visual Studio Code, which makes this obvious, but I thought that maybe Quarto could issue a warning (or info) when encountering unknown chunk options. Currently, I didn't find anything even after setting `--log-level info`. WDYT?</div>

**Comment by cderv** [comment]
As this is encountered from R code here, I believe we can handle that in **knitr**, which does the option parsing. 
Currently, we indeed does not account for the special space and only match with normal space. We could probably support both. 

However, it does not work either for other type of chunk, as with `python` the chunk will still get evaluated.

````markdown
---
title: Test
---

```{python}
#| eval: false
print("Hello")
```
````

We should either: 
* Flag this as invalid and advice to change the space
* Properly extract the option even with non breaking space

**Comment by cbrnr** [comment]
I actually encountered this problem when using the `jupyter` engine, so it would be nice if the solution worked across engines. I would not accept anything other than a regular space, but when encountering `#|` followed by something else, I would issue a warning (or at least an info).

**Comment by cderv** [comment]
@cscheid Could we add that in our YAML validation tooling maybe ?

**Comment by cscheid** [comment]
This is weird. I can see a problem with non-breaking spaces there, but it's not the problem you're seeing. On `main`, I get this:

```
$ quarto render command-option-nbsp.qmd
ERROR: Validation of YAML cell metadata failed.
In file command-option-nbsp.qmd
(line 6, columns 3--8) property name  eval is invalid
5: ```{r}
6: #| eval: false
     ~~~~~
7: print(3 * 4)
```

I see a similar issue with python:

```
$ quarto render command-option-nbsp.qmd
ERROR: Validation of YAML cell metadata failed.
In file command-option-nbsp.qmd
(line 6, columns 3--8) property name  eval is invalid
5: ```{python}
6: #| eval: false
     ~~~~~
7: print("Hello")

ERROR: Render failed due to invalid YAML.
```

I did make some YAML changes recently that probably trigger this, but I think this is actually the correct error message now. (I don't think we should interpret non-breaking spaces as regular spaces)

**Comment by cbrnr** [comment]
I get this error with 1.2.198:

```
quarto-issue  quarto preview test.qmd                         
ERROR: Validation of YAML cell metadata failed.
In file test.qmd
(line 6, columns 4--8) property name  eval is invalid
5: ```{r}
6: #| eval: false
      ~~~~
7: cat("Hello")

ERROR: Render failed due to invalid YAML.
```

So in my case, it only underlines `eval` and *not* the preceding nbsp, which is the reason for the error. But even when all five characters are underlined, I think it will be pretty hard to find out that there is an nbsp instead of a regular space.

**Comment by cscheid** [comment]
Thanks. We've had an inconsistent off-by-one issue with our error underlining code that I haven't been able to isolate. Can you send me your `test.qmd` file so I can try it myself?

**Comment by cbrnr** [comment]
Sure! [test.zip](https://github.com/quarto-dev/quarto-cli/files/9755270/test.zip)

**Comment by cscheid** [comment]
ok, thank you. I can reproduce it. In addition, my previous `command-option-nbsp.qmd` now exhibits the same problem. That means I know what it is; it's a regression caused by us switching the YAML parser we use (we switched it around a few times.) I hope to have a full fix soon.

**Comment by jjallaire** [comment]
@cscheid Is the proposed fix in your opinion safe for v1.2?

**Comment by cscheid** [comment]
@jjallaire Yes, it'll be a local fix on the error reporting. I'll have a PR today.
